lib/chef/resource/chocolatey_source.rb

```diff
@@ -89,22 +90,22 @@ def fetch_source_element(id)
       end
 
       action :enable do
-        introduced "14.6",
+        introduced "14.6"
         description "Enables a Chocolatey source."
 
         if current_resource
-          converge_by("enable Chocolatey source '#{new_resource.source_name}'") do
+          converge_if_changed("enable Chocolatey source '#{new_resource.source_name}'") do
```

Comment:
Sorry that I probably didn't explain this clearly enough yesterday.  For converge_if_changed to work, we need to be retrieving the enabled/disable values when the resource is loaded, have a look at how other fields get retrieved and see if you can work it out, otherwise we can add it together (I'll make it work and you can see how I did it)